### PR TITLE
Added multiple features

### LINKED ISSUE #155: Delete controllers when manager deleted
**Desired Behavior**
MeshSync and broker controllers needs to be deleted when meshery-operator manager gets deleted.

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

- This PR fixes #155 
- Meshkit errors added
- fixed manifest generation with kubebuilder
- Added controller `ClusterRole` with admin permissions

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by dhruv0000 — APPROVED** [review]

File: component_info.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (leecalcote) [inline component_info.json:1]: Gents, are we moving these into a subfolder and not the root directory? // @DelusionalOptimist @mgfeller

**Review by leecalcote — APPROVED** [review]

File: component_info.json (hunk 1 above)
Comment (DelusionalOptimist) [reply component_info.json:1]: Yes, the utility has a new `-o` flag for specifying output directory. We decided upon a new `helpers` directory. Maybe the makefile should be updated in accordance with this.

Comment (leecalcote) [reply component_info.json:1]: Lead the way, @DelusionalOptimist 👍

File: Makefile [hunk 2]
```diff
@@ -65,12 +65,18 @@ fmt:
 vet:
 	go vet ./...
 
+# Run go lint against code
+check:
+	go run github.com/golangci/golangci-lint/cmd/golangci-lint run
+
+# Run meshery error utility against code
+error:
+	go run github.com/layer5io/meshkit/cmd/errorutil -d . update
```
Comment (DelusionalOptimist) [inline Makefile:74]:
```suggestion
	go run github.com/layer5io/meshkit/cmd/errorutil -d . update -i ./helpers -o ./helpers
```
Also move the errorutil files to the new directory

**Review by includestdlibrary — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (kumarabd) [reply Makefile]: @DelusionalOptimist referenced a new issue for you to work on these changes.

**Review by includestdlibrary — APPROVED** [review]

**Pull request merged**